unit-tests/src/test/scala/java/lang/ProcessSuite.scala

```diff
@@ -8,6 +8,20 @@ import scala.scalanative.native.{Zone, toCString}
 import scala.scalanative.posix.{fcntl, unistd}
 import scala.io.Source
 
+// Design Note:
+//     In a number of places waitFor() is called to prevent zombie
+//     processes. These calls have been converted to a timed waitfor.
+//     A timed waitfor means the test will eventually complete, even
+//     if there is a problem with the underlying process.
+//
+//     One minute is an order of magnitude guess for a "reasonable"
+//     completion time.  If a process expected to exit in milliseconds
+//     takes over a minute, it should be reported.
+//
+//     In the expected normal case, the process will exit within seconds
+//     or less, so the 1 minute timeout will not increase the expected
+//     execution time of the test.
+
 object ProcessSuite extends tests.Suite {
   def readInputStream(s: InputStream) = Source.fromInputStream(s).mkString
```

Comment:
Minor suggestion, maybe add the following:
```scala
private implicit class ProcessOs(val process: Process) extends AnyVal {
  def block(): Unit = assert(proc.waitFor(5, TimeUnit.SECONDS), "Process took more than 5 seconds to exit")
}
then you could replace all of the calls to `proc.waitFor` with `proc.block()`. It also allows the time duration to be easily change in one place. None of the these tests should take more than a few hundred milliseconds so I think 5 seconds is plenty of time to wait.
```

Reply:
@eatkins Thank you for the review & suggestion. The PR got better as a result.

I implemented the shorter timeout and a common, central routine.

 I kept the new routine within the ProcessSuite object, rather than your suggested ProcessOs, 
because I think the block or suspend is more closely logically related to the test being
suspended rather than the underlying process.

Please let me know, if possible, if this adequately resolves your concern.  Thank you.

Reply:
@eatkins Thank you, again, for the timely rview.